docs/utilities/data_masking.md

```diff
@@ -117,6 +117,58 @@ Erasing will remove the original data and replace it with a `*****`. This means
     --8<-- "examples/data_masking/src/getting_started_erase_data_output.json"
     ```
 
+### Supported Input Types
+
+You can pass in different types of Python objects. Internally, we convert these to dictionaries for processing.
+
+Examples below show how `erase()` works with each type.
+
+=== "Pydantic Model"
+
+```python
+from pydantic import BaseModel
+from aws_lambda_powertools.utilities.data_masking import DataMasking
+
+class User(BaseModel):
+    name: str
+    age: int
+
+model = User(name="powertools", age=42)
+masked = DataMasking().erase(model, fields=["age"])
+print(masked)  # {'name': 'powertools', 'age': '*****'}
+```
```

Comment:
You can see that we don't add snippets to the `.md` files, but we include a python file in the `examples` folder and then include the snippet here. This way, we can run linter + type checking on our documentation and make sure it works.

Also, make this a Lambda function with entry point `lambda_handler`.